testing.go

```diff
@@ -42,14 +45,18 @@ func (ws *TestWS) Write(t testing.TB, message SocketMessage) {
 	if err != nil {
 		t.Fatalf("could not marshal message before sending to the server %v", err)
 	}
+	ws.writeLock.Lock()
+	defer ws.writeLock.Unlock()
 	if err := ws.WriteMessage(websocket.TextMessage, payloadBytes); err != nil {
 		t.Fatalf("could not send message over ws connection %v", err)
 	}
 }
 
 func (ws *TestWS) readIncomingMessages() {
 	for {
+		ws.readLock.Lock()
 		receivedMessage := &SocketMessage{}
+		ws.readLock.Unlock()
```

Comment:
```suggestion
		receivedMessage := &SocketMessage{}
```
WriteLock is OK, but this readLock is misused, only the call to `ws.ReadJSON` is worth protecting.

However, currently there is only one goroutine `go ws.readIncomingMessages()` doing reads on gorilla websocket - readLock could be completely removed.